low-level/sensors/impl/generic/node_hw.py

```diff
@@ -243,34 +255,31 @@ def initialize(self, conf_reader, msgQlist, products):
         self._bmc_user = Conf.get(GLOBAL_CONF, BMC_USER_KEY, 'ADMIN')
         _bmc_secret = Conf.get(GLOBAL_CONF, BMC_SECRET_KEY, 'ADMIN')
         self._bmc_ip = Conf.get(GLOBAL_CONF, BMC_IP_KEY, '')
-        self._channel_interface = Conf.get(SSPL_CONF,
-            f"{self.BMC_INTERFACE}>{self.BMC_CHANNEL_IF}", 'system')
+
+        self._channel_interface = Conf.get(
+            SSPL_CONF, f"{BMC_INTERFACE}>{BMC_CHANNEL_IF}", 'system')
+        self.active_bmc_if = None
+        self.get_interface_from_cache()
+
         self.iem = Iem()
-        self.iem.check_exsisting_fault_iems()
+        self.iem.check_existing_fault_iems()
         self.IPMI = self.iem.EVENT_CODE["IPMITOOL_AVAILABLE"][1]
-
-        # Decrypt bmc secret
-        decryption_key = encryptor.gen_key(MACHINE_ID,
-            ServiceTypes.SERVER_NODE.value)
-        self.__bmc_passwd = encryptor.decrypt(decryption_key,
-            _bmc_secret, self.SENSOR_NAME)
-        data_dir =  Conf.get(SSPL_CONF, f"{self.SYSINFO}>{self.DATA_PATH_KEY}", self.DATA_PATH_VALUE_DEFAULT)
-        self.cache_dir_path = os.path.join(data_dir, self.CACHE_DIR_NAME)
-
-        # define variable in consul to check for bmc channel interface fallback
-        self.ACTIVE_BMC_IF = os.path.join(self.cache_dir_path, f'ACTIVE_BMC_IF_{self._node_id}')
-        self.LAN_ALERT = os.path.join(self.cache_dir_path, f'LAN_ALERT_{self._node_id}')
-        # save current channel interface value to consul
-        self.active_bmc_if = store.get(self.ACTIVE_BMC_IF)
-        self.lan_fault = store.get(self.LAN_ALERT)
-        if self.active_bmc_if is None:
-            self.active_bmc_if = self._channel_interface
-            store.put(self.active_bmc_if,self.ACTIVE_BMC_IF)
+        if self._channel_interface in BMCInterface.LAN_IF.value:
+            try:
+                # Decrypt bmc secret
+                decryption_key = encryptor.gen_key(
+                    MACHINE_ID, ServiceTypes.SERVER_NODE.value)
+                self._bmc_passwd = encryptor.decrypt(
+                    decryption_key, _bmc_secret, self.SENSOR_NAME)
+            except Exception as err:
+                logger.error(f"BMC password decryption failed due to {err},"
+                    "NodeHWSensor monitoring disabled.")
+                self.shutdown()
 
         # Set flag 'request_shutdown' to true if ipmitool/simulator is non-functional
         _, _, retcode = self._run_ipmitool_subcommand("sel info")
-        if retcode != 0 and self.channel_err is False:
-                self.request_shutdown = True
+        if retcode != 0 and self.channel_err:
+            self.request_shutdown = True
```

Comment:
log.critical with reason.

where is switching to sistem(kcs) interface here

Reply:
>> log.critical with reason.
Done.

>> where is switching to sistem(kcs) interface here
switching to system interface at L728 - where we detecting fault for lan.

Reply:
>"ipmitool commands not able to access BMC through"
                " configured %s interface or even fallback option 'KCS' interface ")

and if configured kcs interface then message should look like 
>"ipmitool commands not able to access BMC through"
                " configured %s interface")

So build message string first based on configured interface and then log it

Reply:
done.